client/src/modules/user/UserProfile/UserProfileName.jsx

```diff
@@ -0,0 +1,164 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { connect } from 'react-redux';
+import _trim from 'lodash/trim';
+import classNames from 'classnames';
+import isEmpty from 'validator/lib/isEmpty';
+import isLength from 'validator/lib/isLength';
+import PropTypes from 'prop-types';
+
+import './UserProfileName.scss';
+import { validateWith } from 'common/utils/helpers';
+import { updateName } from 'modules/user/model/actions';
+
+const UserProfileName = props => {
+  const { name } = props;
+  const [errorMessageId, setErrorMessageId] = useState('');
+  const [isFormVisible, setFormVisibility] = useState(false);
+  const [currentName, setName] = useState(name);
+  const [isDirty, setIsDirty] = useState(false);
+  const nameRef = useRef();
+  const inputRef = useRef();
+
+  const showForm = () => setFormVisibility(true);
+
+  const hideForm = () => setFormVisibility(false);
+
+  const validateName = () => {
+    let errorMessageId;
+
+    errorMessageId = validateWith(
+      value => !isEmpty(value, { ignore_whitespace: true })
+    )('common.form.required-warning')(currentName);
+
+    if (_trim(currentName)) {
+      errorMessageId = validateWith(value =>
+        isLength(value, { min: 1, max: 32 })
+      )('common.form.field-min-max')(currentName);
+    }
+
+    setErrorMessageId(errorMessageId);
+  };
+
+  const handleNameUpdate = async () => {
+    const { userId, updateName } = props;
+
+    try {
+      await updateName(currentName, userId);
+    } catch (error) {
+      /**
+       * Ignore error here as we handle error
+       * via error notification
+       */
+    }
+  };
+
+  const handleSubmit = event => {
+    event.preventDefault();
+
+    if (isDirty && !errorMessageId && isFormVisible) {
+      handleNameUpdate();
+      setErrorMessageId('');
+      setIsDirty(false);
+      hideForm();
+    }
+
+    if (!isDirty && !errorMessageId && isFormVisible) {
+      hideForm();
+    }
+  };
+
+  const handleNameChange = event => {
+    const {
+      target: { value }
+    } = event;
+
+    setName(value);
+    setIsDirty(true);
+  };
+
+  useEffect(() => {
+    validateName();
+  }, [currentName]);
+
+  useEffect(() => {
+    const handleClickOutside = event => {
```

Comment:
please check if this handler is added when input is visible and focused, and if it is removed on blur